## Librerias

In [2]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)
#pd.set_option('display.max_rows', None)
import sys
sys.path.append('C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\utils')
import gold_statistics
from datetime import datetime

## Carga de datos

In [3]:
s10 = pd.read_csv("C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\Data\\raw\\08_19_2020.csv", sep=";")

## Primer análisis

In [4]:
s10

gameId league split  playoffs              date  game  patch  \
0     1241318  KeSPA   NaN         0   03/01/2020 7:33     1   9.24   
1     1241322  KeSPA   NaN         0   03/01/2020 9:00     2   9.24   
2     1241324  KeSPA   NaN         0  03/01/2020 10:05     3   9.24   
3     1241328  KeSPA   NaN         0  03/01/2020 10:57     4   9.24   
4     1241386  KeSPA   NaN         0   04/01/2020 5:33     1   9.24   
...       ...    ...   ...       ...               ...   ...    ...   
5144  1310259     EM   NaN         0  18/08/2020 20:25     1  10.16   
5145  1310260     EM   NaN         0  18/08/2020 20:27     1  10.16   
5146  1310262     EM   NaN         0  18/08/2020 20:53     1  10.16   
5147  1320317     EM   NaN         0  18/08/2020 20:56     1  10.16   
5148  1310264     EM   NaN         0  18/08/2020 21:27     1  10.16   

      gamelength          t1_playerid   t1_ban1       t1_ban2   t1_ban3  \
0           2220       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
1           2227                   T1    Syndra       LeBlanc    Rumble   
2           1711                   T1   LeBlanc        Rumble   Lee Sin   
3           1999                   T1    Rumble         Varus   LeBlanc   
4           1812                  DRX    Qiyana       Orianna    Irelia   
...          ...                  ...       ...           ...       ...   
5144        2075               Racoon   Caitlyn         Karma       Lux   
5145        2095      SINNERS Esports      Shen         Senna      Azir   
5146        2879     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
5147        1874            G2 Arctic  Cho'Gath          Sett  Renekton   
5148        1808  Intrepid Fox Gaming     Akali       Camille     Senna   

           t1_ban4    t1_ban5  t1_result  t1_firstblood  t1_firstdragon  \
0            Yasuo   Renekton          1            0.0             0.0   
1            Braum      Leona          0            0.0             0.0   
2           Gragas       Olaf          1            0.0             1.0   
3           Aatrox     Syndra          0            0.0             1.0   
4       Tahm Kench  Gangplank          0            0.0             0.0   
...            ...        ...        ...            ...             ...   
5144          Shen   Malphite          1            0.0             1.0   
5145  Twisted Fate    Taliyah          1            0.0             0.0   
5146          Olaf       Azir          1            1.0             0.0   
5147   Mordekaiser       Shen          1            0.0             0.0   
5148           Zoe    LeBlanc          1            0.0             1.0   

      t1_dragons  t1_elementaldrakes  t1_infernals  t1_mountains  t1_clouds  \
0              2                 2.0           0.0           2.0        0.0   
1              1                 1.0           0.0           0.0        0.0   
2              4                 4.0           0.0           2.0        1.0   
3              2                 2.0           0.0           1.0        0.0   
4              1                 1.0           0.0           0.0        0.0   
...          ...                 ...           ...           ...        ...   
5144           4                 4.0           2.0           1.0        0.0   
5145           2                 2.0           0.0           0.0        0.0   
5146           4                 4.0           0.0           3.0        0.0   
5147           2                 2.0           1.0           1.0        0.0   
5148           4                 4.0           0.0           2.0        1.0   

      t1_oceans  t1_elders  t1_firstherald  t1_heralds  t1_firstbaron  \
0           0.0        0.0             1.0         1.0            1.0   
1           1.0        0.0             0.0         0.0            0.0   
2           1.0        0.0             1.0         2.0            1.0   
3           1.0        0.0             1.0         1.0            0.0   
4           1.0        0.0            

In [5]:
s10.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5149 entries, 0 to 5148
Data columns (total 408 columns):
 #    Column                        Non-Null Count  Dtype  
---   ------                        --------------  -----  
 0    gameId                        5148 non-null   object 
 1    league                        5149 non-null   object 
 2    split                         3566 non-null   object 
 3    playoffs                      5149 non-null   int64  
 4    date                          5149 non-null   object 
 5    game                          5149 non-null   int64  
 6    patch                         5149 non-null   float64
 7    gamelength                    5149 non-null   int64  
 8    t1_playerid                   5149 non-null   object 
 9    t1_ban1                       4839 non-null   object 
 10   t1_ban2                       4843 non-null   object 
 11   t1_ban3                       5116 non-null   object 
 12   t1_ban4                       4833 non-null   

In [6]:
s10.describe(exclude=object).round(2)

playoffs     game    patch  gamelength  t1_result  t1_firstblood  \
count   5149.00  5149.00  5149.00     5149.00    5149.00        5148.00   
mean       0.09     1.51    10.09     1954.63       0.53           0.51   
std        0.28     0.79     0.06      346.56       0.50           0.50   
min        0.00     1.00     9.24     1005.00       0.00           0.00   
25%        0.00     1.00    10.06     1711.00       0.00           0.00   
50%        0.00     1.00    10.11     1913.00       1.00           1.00   
75%        0.00     2.00    10.13     2158.00       1.00           1.00   
max        1.00     5.00    10.16     3751.00       1.00           1.00   

       t1_firstdragon  t1_dragons  t1_elementaldrakes  t1_infernals  \
count         5095.00     5149.00             5067.00       5067.00   
mean             0.47        2.45                2.32          0.59   
std              0.50        1.48                1.37          0.80   
min              0.00        0.00                0.00          0.00   
25%              0.00        1.00                1.00          0.00   
50%              0.00        3.00                2.00          0.00   
75%              1.00        4.00                4.00          1.00   
max              1.00       10.00                4.00          4.00   

       t1_mountains  t1_clouds  t1_oceans  t1_elders  t1_firstherald  \
count       5067.00    5067.00    5067.00    5098.00         5098.00   
mean           0.57       0.57       0.60       0.08            0.60   
std            0.81       0.79       0.82       0.28            0.49   
min            0.00       0.00       0.00       0.00            0.00   
25%            0.00       0.00       0.00       0.00            0.00   
50%            0.00       0.00       0.00       0.00            1.00   
75%            1.00       1.00       1.00       0.00            1.00   
max            4.00       4.00       4.00       2.00            1.00   

       t1_heralds  t1_firstbaron  t1_barons  t1_firsttower  t1_towers  \
count     5015.00        4964.00    5149.00        5119.00    5149.00   
mean         1.18           0.49       0.64           0.57       6.23   
std          0.77           0.50       0.69           0.50       3.58   
min          0.00           0.00       0.00           0.00       0.00   
25%          1.00           0.00       0.00           0.00       3.00   
50%          1.00           0.00       1.00           1.00       7.00   
75%          2.00           1.00       1.00           1.00       9.00   
max          2.00           1.00       4.00           1.00      11.00   

       t1_firstmidtower  t1_firsttothreetowers  t1_inhibitors  t2_result  \
count           5091.00                5098.00        5098.00    5149.00   
mean               0.57                   0.56           1.00       0.47   
std                0.50                   0.50           1.12       0.50   
min                0.00                   0.00           0.00       0.00   
25%                0.00                   0.00           0.00       0.00   
50%                1.00                   1.00           1.00       0.00   
75%                1.00                   1.00           2.00       1.00   
max                1.00                   1.00           7.00       1.00   

       t2_firstblood  t2_firstdragon  t2_dragons  t2_elementaldrakes  \
count        5148.00         5095.00     5149.00             5067.00   
mean            0.49            0.53        2.45                2.31   
std             0.50            0.50        1.55                1.39   
min             0.00            0.00        0.00                0.00   
25%             0.00            0.00        1.00                1.00   
50%             0.00            1.00        2.00                2.00   
75%             1.00            1.00        4.00                4.00   
max             1.00            1.00       10.00                4.00   

       t2_infernals  t2_mountains  t2_clouds  t2_o

In [7]:
s10.describe(exclude=object).shape

(8, 352)

In [8]:
s10.describe(include=object)

gameId league   split              date t1_playerid  t1_ban1 t1_ban2  \
count      5148   5149    3566              5149        5149     4839    4843   
unique     5148     23       7              5027         244      107     109   
top     1241318    LDL  Spring  06/04/2020 19:28         DRX  Kalista    Sett   
freq          1    840    1522                 4          59      314     266   

       t1_ban3  t1_ban4  t1_ban5      t2_playerid t2_ban1 t2_ban2 t2_ban3  \
count     5116     4833     4821             5149    5115    5118    5111   
unique     111      128      124              244     101     105     106   
top       Sett  LeBlanc  LeBlanc  FunPlus Phoenix   Varus    Sett    Sett   
freq       348      286      230               59     810     427     441   

        t2_ban4  t2_ban5 t1p1_position t1p1_player t1p1_team t1p1_champion  \
count      5102     5090          5149        5149      5149          5149   
unique      135      125             1         341       244            51   
top     LeBlanc  LeBlanc           sup       Keria       DRX      Nautilus   
freq        268      243          5149          59        59           946   

       t1p2_position t1p2_player t1p2_team t1p2_champion t1p3_position  \
count           5149        5149      5149          5149          5149   
unique             1         331       244            54             1   
top              bot        Deft       DRX      Aphelios           mid   
freq            5149          59        59          1184          5149   

       t1p3_player t1p3_team t1p3_champion t1p4_position t1p4_player  \
count         5149      5149          5149          5149        5149   
unique         349       244            83             1         335   
top          Chovy       DRX        Syndra           jng      Pyosik   
freq            57        59           555          5149          59   

       t1p4_team t1p4_champion t1p5_position t1p5_player t1p5_team  \
count       5149          5149          5149        5149      5149   
unique       244            47             1         339       244   
top          DRX       Lee Sin           top       Doran       DRX   
freq          59           720          5149          57        59   

       t1p5_champion t2p1_position t2p1_player        t2p1_team t2p1_champion  \
count           5149          5149        5149             5149          5149   
unique            72             1         347              244            55   
top             Ornn           sup      Shadow  FunPlus Phoenix      Nautilus   
freq             735          5149          60               59           930   

       t2p2_position t2p2_player        t2p2_team t2p2_champion t2p3_position  \
count           5149        5149             5149          5149          5149   
unique             1         335              244            54             1   
top              bot         Lwx  FunPlus Phoenix      Aphelios           mid   
freq            5149          58               59           910          5149   

       t2p3_player        t2p3_team t2p3_champion t2p4_position t2p4_player  \
count         5149             5149          5149          5149        5149   
unique         344              244            87             1         331   
top          Doinb  FunPlus Phoenix           Zoe           jng        Tian   
freq            59               59           529          5149          59   

              t2p4_team t2p4_champion t2p5_position t2p5_player  \
count              5149          5149          5149        5149   
unique              244            47             1         334   
top     FunPlus Phoenix       Lee Sin           top  StarScreen   
freq                 59           778          5149          50   

              t2p5_team t2p5_champion  
count              5149          5149  
unique              244            70  
top     FunPlus Phoenix          Ornn  
freq                 59           727

In [9]:
s10.describe(include=object).shape

(4, 56)

In [10]:
s10.duplicated().sum()

0

In [11]:
for i in s10.isna().sum()/len(s10)*100:
    print(i)

0.019421246844047387
0.0
30.743833754127014
0.0
0.0
0.0
0.0
0.0
0.0
6.02058652165469
5.9429015342785005
0.6409011458535638
6.137114002718975
6.370168964847543
0.0
0.019421246844047387
1.048747329578559
0.0
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
0.9904835890464169
0.9904835890464169
2.60244707710235
3.5929306661487663
0.0
0.5826374053214216
0.0
1.1264323169547485
0.9904835890464169
0.9904835890464169
0.0
0.6603223926976112
0.602058652165469
0.7380073800738007
0.9127986016702273
1.1458535637987959
0.0
0.019421246844047387
1.048747329578559
0.0
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
0.9904835890464169
0.9904835890464169
2.60244707710235
3.5929306661487663
0.0
0.5826374053214216
0.0
1.1264323169547485
0.9904835890464169
0.9904835890464169
0.0
0.0
0.0
0.0
0.0
0.0
0.0
2.0392309186249755
0.602058652165469
1.009904835890464
1.009904835890464
0.019421246844047387
0.019421246844047387


Tras este primer análisis inicial, el dataset consta de 5149 fias y 408 columnas y ninguna fila con datos incoherentes tanto en el encabezado como en la cola. Se observa que la mayoría de las columnas poseen valores nulos (``Nan``), pero es cierto que ninguna de estas supera el 2% de valores nulos en su interior. Los tipo de dato de las columnas se reparten en 352 numéricas (241 de tipo ``float`` y 61 de tipo ``int``). el tipo de los datos de las 56 columnas restantes es tipo ``objeto``. Además, la información aportada por la función ``describe`` de ``pandas`` nos permite intuir que los datos no siguen una distribución normal aparente. Esta misma función nos puede dar a conocer variables binomiales como ``t1_result`` pues su desviación típica, máximo y mínimo se asemejana a como son los de balores booleanos. Por último, destacamos que no existen columnas duplicadas. 

## Primer tratamiento del set de datos

Antes de seguir con el análisis, vamos a hacer un tratamiento previo del ``DataFrame``. Debido a conocomiento de previo y lectura de la descripción de los datos, vamos a quedarnos solo con las columnas con datos sobre el equipo y no de los jugadores en específico pues de momento nos centraremos en ello.  

Pasando de *t1* a *blue*

In [12]:
s10.columns[s10.columns.str.startswith("t1")]

Index(['t1_playerid', 't1_ban1', 't1_ban2', 't1_ban3', 't1_ban4', 't1_ban5',
       't1_result', 't1_firstblood', 't1_firstdragon', 't1_dragons',
       ...
       't1p5_monsterkills', 't1p5_monsterkillsownjungle',
       't1p5_monsterkillsenemyjungle', 't1p5_cspm', 't1p5_goldat10',
       't1p5pat10', 't1p5_csat10', 't1p5_goldat15', 't1p5pat15',
       't1p5_csat15'],
      dtype='object', length=200)

In [13]:
s10.columns = s10.columns.str.replace("t1_", "blue_")

In [14]:
# t1p1
s10.columns = s10.columns.str.replace("t1p1", "blue_p1")

# t1p2
s10.columns = s10.columns.str.replace("t1p2", "blue_p2")

#t1p3
s10.columns = s10.columns.str.replace("t1p3", "blue_p3")

#t1p4

s10.columns = s10.columns.str.replace("t1p4", "blue_p4")

#t1p4

s10.columns = s10.columns.str.replace("t1p5", "blue_p5")

In [15]:
s10.columns[s10.columns.str.startswith("blue")]

Index(['blue_playerid', 'blue_ban1', 'blue_ban2', 'blue_ban3', 'blue_ban4',
       'blue_ban5', 'blue_result', 'blue_firstblood', 'blue_firstdragon',
       'blue_dragons',
       ...
       'blue_p5_monsterkills', 'blue_p5_monsterkillsownjungle',
       'blue_p5_monsterkillsenemyjungle', 'blue_p5_cspm', 'blue_p5_goldat10',
       'blue_p5pat10', 'blue_p5_csat10', 'blue_p5_goldat15', 'blue_p5pat15',
       'blue_p5_csat15'],
      dtype='object', length=200)

Pasando de *t2* a *red*

In [16]:
s10.columns[s10.columns.str.startswith("t2")]

Index(['t2_playerid', 't2_ban1', 't2_ban2', 't2_ban3', 't2_ban4', 't2_ban5',
       't2_result', 't2_firstblood', 't2_firstdragon', 't2_dragons',
       ...
       't2p5_monsterkills', 't2p5_monsterkillsownjungle',
       't2p5_monsterkillsenemyjungle', 't2p5_cspm', 't2p5_goldat10',
       't2p5pat10', 't2p5_csat10', 't2p5_goldat15', 't2p5pat15',
       't2p5_csat15'],
      dtype='object', length=200)

In [17]:
# De t2 a red 

s10.columns = s10.columns.str.replace("t2_", "red_")

s10.columns = s10.columns.str.replace("t2", "red_")

In [18]:
s10.columns[s10.columns.str.startswith("red")]

Index(['red_playerid', 'red_ban1', 'red_ban2', 'red_ban3', 'red_ban4',
       'red_ban5', 'red_result', 'red_firstblood', 'red_firstdragon',
       'red_dragons',
       ...
       'red_p5_monsterkills', 'red_p5_monsterkillsownjungle',
       'red_p5_monsterkillsenemyjungle', 'red_p5_cspm', 'red_p5_goldat10',
       'red_p5pat10', 'red_p5_csat10', 'red_p5_goldat15', 'red_p5pat15',
       'red_p5_csat15'],
      dtype='object', length=200)

columnas de posición

In [19]:
s10[s10.columns[s10.columns.str.endswith("position")]]

blue_p1_position blue_p2_position blue_p3_position blue_p4_position  \
0                 sup              bot              mid              jng   
1                 sup              bot              mid              jng   
2                 sup              bot              mid              jng   
3                 sup              bot              mid              jng   
4                 sup              bot              mid              jng   
...               ...              ...              ...              ...   
5144              sup              bot              mid              jng   
5145              sup              bot              mid              jng   
5146              sup              bot              mid              jng   
5147              sup              bot              mid              jng   
5148              sup              bot              mid              jng   

     blue_p5_position red_p1_position red_p2_position red_p3_position  \
0                 top             sup             bot             mid   
1                 top             sup             bot             mid   
2                 top             sup             bot             mid   
3                 top             sup             bot             mid   
4                 top             sup             bot             mid   
...               ...             ...             ...             ...   
5144              top             sup             bot             mid   
5145              top             sup             bot             mid   
5146              top             sup             bot             mid   
5147              top             sup             bot             mid   
5148              top             sup             bot             mid   

     red_p4_position red_p5_position  
0                jng             top  
1                jng             top  
2                jng             top  
3                jng             top  
4                jng             top  
...              ...             ...  
5144             jng             top  
5145             jng             top  
5146             jng             top  
5147             jng             top  
5148             jng             top  

[5149 rows x 10 columns]

In [20]:
s10[s10.columns[s10.columns.str.endswith("position")]].value_counts()

blue_p1_position  blue_p2_position  blue_p3_position  blue_p4_position  blue_p5_position  red_p1_position  red_p2_position  red_p3_position  red_p4_position  red_p5_position
sup               bot               mid               jng               top               sup              bot              mid              jng              top                5149
Name: count, dtype: int64

In [21]:
# p1 to sup
s10.columns = s10.columns.str.replace("p1", "sup")

#p2 to bot
s10.columns = s10.columns.str.replace("p2", "bot")

#p3 to mid
s10.columns = s10.columns.str.replace("p3", "mid")

#p4 to jng
s10.columns = s10.columns.str.replace("p4", "jng")

#p5 to top
s10.columns = s10.columns.str.replace("p5", "top")

In [22]:
position_cols = s10[s10.columns[s10.columns.str.endswith("position")]]

for i in s10:
    if i in position_cols:
        del s10[f"{i}"]

Trabajando con las columnas de equipos

In [23]:
s10[s10.columns[s10.columns.str.endswith("playerid")]]

blue_playerid          red_playerid
0          SANDBOX Gaming                    T1
1                      T1        SANDBOX Gaming
2                      T1        SANDBOX Gaming
3                      T1        SANDBOX Gaming
4                     DRX        Afreeca Freecs
...                   ...                   ...
5144               Racoon  SAMCLAN Esports Club
5145      SINNERS Esports            Five Kings
5146     Team GamerLegion        Riddle Esports
5147            G2 Arctic                 CR4ZY
5148  Intrepid Fox Gaming     7more7 Pompa Team

[5149 rows x 2 columns]

In [24]:
s10[s10.columns[s10.columns.str.endswith("playerid")]].nunique()

blue_playerid    244
red_playerid     244
dtype: int64

In [25]:
s10[s10.columns[s10.columns.str.endswith("team")]]

blue_sup_team        blue_bot_team        blue_mid_team  \
0          SANDBOX Gaming       SANDBOX Gaming       SANDBOX Gaming   
1                      T1                   T1                   T1   
2                      T1                   T1                   T1   
3                      T1                   T1                   T1   
4                     DRX                  DRX                  DRX   
...                   ...                  ...                  ...   
5144               Racoon               Racoon               Racoon   
5145      SINNERS Esports      SINNERS Esports      SINNERS Esports   
5146     Team GamerLegion     Team GamerLegion     Team GamerLegion   
5147            G2 Arctic            G2 Arctic            G2 Arctic   
5148  Intrepid Fox Gaming  Intrepid Fox Gaming  Intrepid Fox Gaming   

            blue_jng_team        blue_top_team          red_sup_team  \
0          SANDBOX Gaming       SANDBOX Gaming                    T1   
1                      T1                   T1        SANDBOX Gaming   
2                      T1                   T1        SANDBOX Gaming   
3                      T1                   T1        SANDBOX Gaming   
4                     DRX                  DRX        Afreeca Freecs   
...                   ...                  ...                   ...   
5144               Racoon               Racoon  SAMCLAN Esports Club   
5145      SINNERS Esports      SINNERS Esports            Five Kings   
5146     Team GamerLegion     Team GamerLegion        Riddle Esports   
5147            G2 Arctic            G2 Arctic                 CR4ZY   
5148  Intrepid Fox Gaming  Intrepid Fox Gaming     7more7 Pompa Team   

              red_bot_team          red_mid_team          red_jng_team  \
0                       T1                    T1                    T1   
1           SANDBOX Gaming        SANDBOX Gaming        SANDBOX Gaming   
2           SANDBOX Gaming        SANDBOX Gaming        SANDBOX Gaming   
3           SANDBOX Gaming        SANDBOX Gaming        SANDBOX Gaming   
4           Afreeca Freecs        Afreeca Freecs        Afreeca Freecs   
...                    ...                   ...                   ...   
5144  SAMCLAN Esports Club  SAMCLAN Esports Club  SAMCLAN Esports Club   
5145            Five Kings            Five Kings            Five Kings   
5146        Riddle Esports        Riddle Esports        Riddle Esports   
5147                 CR4ZY                 CR4ZY                 CR4ZY   
5148     7more7 Pompa Team     7more7 Pompa Team     7more7 Pompa Team   

              red_top_team  
0                       T1  
1           SANDBOX Gaming  
2           SANDBOX Gaming  
3           SANDBOX Gaming  
4           Afreeca Freecs  
...                    ...  
5144  SAMCLAN Esports Club  
5145            Five Kings  
5146        Riddle Esports  
5147                 CR4ZY  
5148     7more7 Pompa Team  

[5149 rows x 10 columns]

In [26]:
s10[s10.columns[s10.columns.str.endswith("playerid")]].nunique()

blue_playerid    244
red_playerid     244
dtype: int64

In [27]:
b_team = s10[["blue_top_team", "blue_mid_team", "blue_jng_team", "blue_bot_team", "blue_sup_team"]]
r_team = s10[["red_top_team", "red_mid_team", "red_jng_team", "red_bot_team", "red_sup_team"]]

for team_col in s10:
    if team_col in b_team:
        del s10[f"{team_col}"]
        
    elif team_col in r_team:
        del s10[f"{team_col}"]

Columna victorias (columna target)

In [28]:
s10["blue_result"].value_counts()

blue_result
1    2725
0    2424
Name: count, dtype: int64

In [29]:
s10["red_result"].value_counts()

red_result
0    2725
1    2424
Name: count, dtype: int64

In [30]:
s10.columns = s10.columns.str.replace("blue_result", "result")

In [31]:
del s10["red_result"]

Primeros objetivos

In [32]:
s10.columns[s10.columns.str.contains('first')]

Index(['blue_firstblood', 'blue_firstdragon', 'blue_firstherald',
       'blue_firstbaron', 'blue_firsttower', 'blue_firstmidtower',
       'blue_firsttothreetowers', 'red_firstblood', 'red_firstdragon',
       'red_firstherald', 'red_firstbaron', 'red_firsttower',
       'red_firstmidtower', 'red_firsttothreetowers', 'blue_sup_firstblood',
       'blue_sup_firstbloodkill', 'blue_sup_firstbloodassist',
       'blue_sup_firstbloodvictim', 'blue_bot_firstblood',
       'blue_bot_firstbloodkill', 'blue_bot_firstbloodassist',
       'blue_bot_firstbloodvictim', 'blue_mid_firstblood',
       'blue_mid_firstbloodkill', 'blue_mid_firstbloodassist',
       'blue_mid_firstbloodvictim', 'blue_jng_firstblood',
       'blue_jng_firstbloodkill', 'blue_jng_firstbloodassist',
       'blue_jng_firstbloodvictim', 'blue_top_firstblood',
       'blue_top_firstbloodkill', 'blue_top_firstbloodassist',
       'blue_top_firstbloodvictim', 'red_sup_firstblood',
       'red_sup_firstbloodkill', 'red_sup_firstb

In [33]:
s10.columns =s10.columns.str.replace("blue_first", "first")

In [34]:
red_first = s10[s10.columns[s10.columns.str.startswith('red_first')]]

for  x in s10:
    if x in red_first:
        del s10[f"{x}"]

In [35]:
firstbloods  = s10.columns[s10.columns.str.endswith('_firstblood')]

for y in s10:
    if y in firstbloods:
        del s10[f"{y}"]

In [36]:
s10.columns[s10.columns.str.contains('first')]

Index(['firstblood', 'firstdragon', 'firstherald', 'firstbaron', 'firsttower',
       'firstmidtower', 'firsttothreetowers', 'blue_sup_firstbloodkill',
       'blue_sup_firstbloodassist', 'blue_sup_firstbloodvictim',
       'blue_bot_firstbloodkill', 'blue_bot_firstbloodassist',
       'blue_bot_firstbloodvictim', 'blue_mid_firstbloodkill',
       'blue_mid_firstbloodassist', 'blue_mid_firstbloodvictim',
       'blue_jng_firstbloodkill', 'blue_jng_firstbloodassist',
       'blue_jng_firstbloodvictim', 'blue_top_firstbloodkill',
       'blue_top_firstbloodassist', 'blue_top_firstbloodvictim',
       'red_sup_firstbloodkill', 'red_sup_firstbloodassist',
       'red_sup_firstbloodvictim', 'red_bot_firstbloodkill',
       'red_bot_firstbloodassist', 'red_bot_firstbloodvictim',
       'red_mid_firstbloodkill', 'red_mid_firstbloodassist',
       'red_mid_firstbloodvictim', 'red_jng_firstbloodkill',
       'red_jng_firstbloodassist', 'red_jng_firstbloodvictim',
       'red_top_firstbloodkill'

Estadisticas de oro por equipo

In [37]:
s10.columns[s10.columns.str.contains("gold")]

Index(['blue_sup_totalgold', 'blue_sup_earnedgold', 'blue_sup_goldspent',
       'blue_sup_goldat10', 'blue_sup_goldat15', 'blue_bot_totalgold',
       'blue_bot_earnedgold', 'blue_bot_goldspent', 'blue_bot_goldat10',
       'blue_bot_goldat15', 'blue_mid_totalgold', 'blue_mid_earnedgold',
       'blue_mid_goldspent', 'blue_mid_goldat10', 'blue_mid_goldat15',
       'blue_jng_totalgold', 'blue_jng_earnedgold', 'blue_jng_goldspent',
       'blue_jng_goldat10', 'blue_jng_goldat15', 'blue_top_totalgold',
       'blue_top_earnedgold', 'blue_top_goldspent', 'blue_top_goldat10',
       'blue_top_goldat15', 'red_sup_totalgold', 'red_sup_earnedgold',
       'red_sup_goldspent', 'red_sup_goldat10', 'red_sup_goldat15',
       'red_bot_totalgold', 'red_bot_earnedgold', 'red_bot_goldspent',
       'red_bot_goldat10', 'red_bot_goldat15', 'red_mid_totalgold',
       'red_mid_earnedgold', 'red_mid_goldspent', 'red_mid_goldat10',
       'red_mid_goldat15', 'red_jng_totalgold', 'red_jng_earnedgold',
  

In [38]:
gold_stats = s10[s10.columns[s10.columns.str.contains("gold")]]

#lado azul

blue_gold_stats = gold_stats[gold_stats.columns[gold_stats.columns.str.contains("blue")]]

#total
s10["blue_totalgold"] = blue_gold_stats[blue_gold_stats.columns[blue_gold_stats.columns.str.contains("total")]].sum(axis=1)

#earned
s10["blue_earnedgold"] = blue_gold_stats[blue_gold_stats.columns[blue_gold_stats.columns.str.contains("earned")]].sum(axis=1)

#spent
s10["blue_spentgold"] = blue_gold_stats[blue_gold_stats.columns[blue_gold_stats.columns.str.contains("spent")]].sum(axis=1)

#at10
s10["blue_goldat10"] = blue_gold_stats[blue_gold_stats.columns[blue_gold_stats.columns.str.contains("at10")]].sum(axis=1)


#at15
s10["blue_goldat15"] = blue_gold_stats[blue_gold_stats.columns[blue_gold_stats.columns.str.contains("at15")]].sum(axis=1)


In [39]:
red_gold_stats = gold_stats[gold_stats.columns[gold_stats.columns.str.contains("red")]]

#total
s10["red_totalgold"] = red_gold_stats[red_gold_stats.columns[red_gold_stats.columns.str.contains("total")]].sum(axis=1)

#earned
s10["red_earnedgold"] = red_gold_stats[red_gold_stats.columns[red_gold_stats.columns.str.contains("earned")]].sum(axis=1)

#spent
s10["red_spentgold"] = red_gold_stats[red_gold_stats.columns[red_gold_stats.columns.str.contains("spent")]].sum(axis=1)

#at10
s10["red_goldat10"] = red_gold_stats[red_gold_stats.columns[red_gold_stats.columns.str.contains("at10")]].sum(axis=1)


#at15
s10["red_goldat15"] = red_gold_stats[red_gold_stats.columns[red_gold_stats.columns.str.contains("at15")]].sum(axis=1)

In [40]:
gold_statistics.get_gold_diff(s10)

gameId league split  playoffs              date  game  patch  \
0     1241318  KeSPA   NaN         0   03/01/2020 7:33     1   9.24   
1     1241322  KeSPA   NaN         0   03/01/2020 9:00     2   9.24   
2     1241324  KeSPA   NaN         0  03/01/2020 10:05     3   9.24   
3     1241328  KeSPA   NaN         0  03/01/2020 10:57     4   9.24   
4     1241386  KeSPA   NaN         0   04/01/2020 5:33     1   9.24   
...       ...    ...   ...       ...               ...   ...    ...   
5144  1310259     EM   NaN         0  18/08/2020 20:25     1  10.16   
5145  1310260     EM   NaN         0  18/08/2020 20:27     1  10.16   
5146  1310262     EM   NaN         0  18/08/2020 20:53     1  10.16   
5147  1320317     EM   NaN         0  18/08/2020 20:56     1  10.16   
5148  1310264     EM   NaN         0  18/08/2020 21:27     1  10.16   

      gamelength        blue_playerid blue_ban1     blue_ban2 blue_ban3  \
0           2220       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
1           2227                   T1    Syndra       LeBlanc    Rumble   
2           1711                   T1   LeBlanc        Rumble   Lee Sin   
3           1999                   T1    Rumble         Varus   LeBlanc   
4           1812                  DRX    Qiyana       Orianna    Irelia   
...          ...                  ...       ...           ...       ...   
5144        2075               Racoon   Caitlyn         Karma       Lux   
5145        2095      SINNERS Esports      Shen         Senna      Azir   
5146        2879     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
5147        1874            G2 Arctic  Cho'Gath          Sett  Renekton   
5148        1808  Intrepid Fox Gaming     Akali       Camille     Senna   

         blue_ban4  blue_ban5  result  firstblood  firstdragon  blue_dragons  \
0            Yasuo   Renekton       1         0.0          0.0             2   
1            Braum      Leona       0         0.0          0.0             1   
2           Gragas       Olaf       1         0.0          1.0             4   
3           Aatrox     Syndra       0         0.0          1.0             2   
4       Tahm Kench  Gangplank       0         0.0          0.0             1   
...            ...        ...     ...         ...          ...           ...   
5144          Shen   Malphite       1         0.0          1.0             4   
5145  Twisted Fate    Taliyah       1         0.0          0.0             2   
5146          Olaf       Azir       1         1.0          0.0             4   
5147   Mordekaiser       Shen       1         0.0          0.0             2   
5148           Zoe    LeBlanc       1         0.0          1.0             4   

      blue_elementaldrakes  blue_infernals  blue_mountains  blue_clouds  \
0                      2.0             0.0             2.0          0.0   
1                      1.0             0.0             0.0          0.0   
2                      4.0             0.0             2.0          1.0   
3                      2.0             0.0             1.0          0.0   
4                      1.0             0.0             0.0          0.0   
...                    ...             ...             ...          ...   
5144                   4.0             2.0             1.0          0.0   
5145                   2.0             0.0             0.0          0.0   
5146                   4.0             0.0             3.0          0.0   
5147                   2.0             1.0             1.0          0.0   
5148                   4.0             0.0             2.0          1.0   

      blue_oceans  blue_elders  firstherald  blue_heralds  firstbaron  \
0             0.0          0.0          1.0           1.0         1.0   
1             1.0          0.0          0.0           0.0         0.0   
2             1.0          0.0          1.0           2.0         1.0   
3             1.0          0.0          1.0           1.0         0.0   
4             1.0         

date to datetime

In [41]:
s10["date"].dtypes

dtype('O')

In [42]:
s10["date"] = pd.to_datetime(s10["date"], dayfirst= True, format="mixed")

In [43]:
s10["date"].dtypes

dtype('<M8[ns]')

In [44]:
s10.set_index("date")

gameId league split  playoffs  game  patch  gamelength  \
date                                                                           
2020-01-03 07:33:00  1241318  KeSPA   NaN         0     1   9.24        2220   
2020-01-03 09:00:00  1241322  KeSPA   NaN         0     2   9.24        2227   
2020-01-03 10:05:00  1241324  KeSPA   NaN         0     3   9.24        1711   
2020-01-03 10:57:00  1241328  KeSPA   NaN         0     4   9.24        1999   
2020-01-04 05:33:00  1241386  KeSPA   NaN         0     1   9.24        1812   
...                      ...    ...   ...       ...   ...    ...         ...   
2020-08-18 20:25:00  1310259     EM   NaN         0     1  10.16        2075   
2020-08-18 20:27:00  1310260     EM   NaN         0     1  10.16        2095   
2020-08-18 20:53:00  1310262     EM   NaN         0     1  10.16        2879   
2020-08-18 20:56:00  1320317     EM   NaN         0     1  10.16        1874   
2020-08-18 21:27:00  1310264     EM   NaN         0     1  10.16        1808   

                           blue_playerid blue_ban1     blue_ban2 blue_ban3  \
date                                                                         
2020-01-03 07:33:00       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
2020-01-03 09:00:00                   T1    Syndra       LeBlanc    Rumble   
2020-01-03 10:05:00                   T1   LeBlanc        Rumble   Lee Sin   
2020-01-03 10:57:00                   T1    Rumble         Varus   LeBlanc   
2020-01-04 05:33:00                  DRX    Qiyana       Orianna    Irelia   
...                                  ...       ...           ...       ...   
2020-08-18 20:25:00               Racoon   Caitlyn         Karma       Lux   
2020-08-18 20:27:00      SINNERS Esports      Shen         Senna      Azir   
2020-08-18 20:53:00     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
2020-08-18 20:56:00            G2 Arctic  Cho'Gath          Sett  Renekton   
2020-08-18 21:27:00  Intrepid Fox Gaming     Akali       Camille     Senna   

                        blue_ban4  blue_ban5  result  firstblood  firstdragon  \
date                                                                            
2020-01-03 07:33:00         Yasuo   Renekton       1         0.0          0.0   
2020-01-03 09:00:00         Braum      Leona       0         0.0          0.0   
2020-01-03 10:05:00        Gragas       Olaf       1         0.0          1.0   
2020-01-03 10:57:00        Aatrox     Syndra       0         0.0          1.0   
2020-01-04 05:33:00    Tahm Kench  Gangplank       0         0.0          0.0   
...                           ...        ...     ...         ...          ...   
2020-08-18 20:25:00          Shen   Malphite       1         0.0          1.0   
2020-08-18 20:27:00  Twisted Fate    Taliyah       1         0.0          0.0   
2020-08-18 20:53:00          Olaf       Azir       1         1.0          0.0   
2020-08-18 20:56:00   Mordekaiser       Shen       1         0.0          0.0   
2020-08-18 21:27:00           Zoe    LeBlanc       1         0.0          1.0   

                     blue_dragons  blue_elementaldrakes  blue_infernals  \
date                                                                      
2020-01-03 07:33:00             2                   2.0             0.0   
2020-01-03 09:00:00             1                   1.0             0.0   
2020-01-03 10:05:00             4                   4.0             0.0   
2020-01-03 10:57:00             2                   2.0             0.0   
2020-01-04 05:33:00             1                   1.0             0.0   
...                           ...                   ...             ...   
2020-08-18 20:25:00             4                   4.0             2.0   
2020-08-18 20:27:00             2                   2.0             0.0   
2020-08-18 20:53:00             4                   4.0             0.0   
2020-08-18 20:56:00             2                   2.0             1.0   
2020-08-18 2

Procedemos guardar el ``DataFrame`` en un csv, ya que en un futuro a partir de este punto podríamos seguir adelante con otras columnas target y será más cómodo a partir de este punto

In [45]:
s10.to_csv("C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\Data\\process\\s10_0.csv", index=False)

## Eliminando columnas jugador

In [46]:
#cargamos los datos 
s10 = pd.read_csv("C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\Data\\process\\s10_0.csv", index_col=False, parse_dates=["date"])

In [47]:
s10

gameId league split  playoffs                date  game  patch  \
0     1241318  KeSPA   NaN         0 2020-01-03 07:33:00     1   9.24   
1     1241322  KeSPA   NaN         0 2020-01-03 09:00:00     2   9.24   
2     1241324  KeSPA   NaN         0 2020-01-03 10:05:00     3   9.24   
3     1241328  KeSPA   NaN         0 2020-01-03 10:57:00     4   9.24   
4     1241386  KeSPA   NaN         0 2020-01-04 05:33:00     1   9.24   
...       ...    ...   ...       ...                 ...   ...    ...   
5144  1310259     EM   NaN         0 2020-08-18 20:25:00     1  10.16   
5145  1310260     EM   NaN         0 2020-08-18 20:27:00     1  10.16   
5146  1310262     EM   NaN         0 2020-08-18 20:53:00     1  10.16   
5147  1320317     EM   NaN         0 2020-08-18 20:56:00     1  10.16   
5148  1310264     EM   NaN         0 2020-08-18 21:27:00     1  10.16   

      gamelength        blue_playerid blue_ban1     blue_ban2 blue_ban3  \
0           2220       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
1           2227                   T1    Syndra       LeBlanc    Rumble   
2           1711                   T1   LeBlanc        Rumble   Lee Sin   
3           1999                   T1    Rumble         Varus   LeBlanc   
4           1812                  DRX    Qiyana       Orianna    Irelia   
...          ...                  ...       ...           ...       ...   
5144        2075               Racoon   Caitlyn         Karma       Lux   
5145        2095      SINNERS Esports      Shen         Senna      Azir   
5146        2879     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
5147        1874            G2 Arctic  Cho'Gath          Sett  Renekton   
5148        1808  Intrepid Fox Gaming     Akali       Camille     Senna   

         blue_ban4  blue_ban5  result  firstblood  firstdragon  blue_dragons  \
0            Yasuo   Renekton       1         0.0          0.0             2   
1            Braum      Leona       0         0.0          0.0             1   
2           Gragas       Olaf       1         0.0          1.0             4   
3           Aatrox     Syndra       0         0.0          1.0             2   
4       Tahm Kench  Gangplank       0         0.0          0.0             1   
...            ...        ...     ...         ...          ...           ...   
5144          Shen   Malphite       1         0.0          1.0             4   
5145  Twisted Fate    Taliyah       1         0.0          0.0             2   
5146          Olaf       Azir       1         1.0          0.0             4   
5147   Mordekaiser       Shen       1         0.0          0.0             2   
5148           Zoe    LeBlanc       1         0.0          1.0             4   

      blue_elementaldrakes  blue_infernals  blue_mountains  blue_clouds  \
0                      2.0             0.0             2.0          0.0   
1                      1.0             0.0             0.0          0.0   
2                      4.0             0.0             2.0          1.0   
3                      2.0             0.0             1.0          0.0   
4                      1.0             0.0             0.0          0.0   
...                    ...             ...             ...          ...   
5144                   4.0             2.0             1.0          0.0   
5145                   2.0             0.0             0.0          0.0   
5146                   4.0             0.0             3.0          0.0   
5147                   2.0             1.0             1.0          0.0   
5148                   4.0             0.0             2.0          1.0   

      blue_oceans  blue_elders  firstherald  blue_heralds  firstbaron  \
0             0.0          0.0          1.0           1.0         1.0   
1             1.0          0.0          0.0           0.0         0.0   
2             1.0          0.0          1.0           2.0         1.0   
3             1.0          0.0          1.0           1.0         0.0   
4 

In [48]:
support = s10[s10.columns[s10.columns.str.contains("sup")]]

for sup in s10:
    if sup in support:
        del s10[f"{sup}"]

In [49]:
adc = s10[s10.columns[s10.columns.str.contains("bot")]]

for bot in s10:
    if bot in adc:
        del s10[f"{bot}"]

In [50]:
midlane = s10[s10.columns[s10.columns.str.contains("_mid")]]

for mid in s10:
    if mid in midlane:
        del s10[f"{mid}"]

In [51]:
jungler = s10[s10.columns[s10.columns.str.contains("jng")]]

for jng in s10:
    if jng in jungler:
        del s10[f"{jng}"]

In [52]:
toplane = s10[s10.columns[s10.columns.str.contains("top")]]

for top in s10:
    if top in toplane:
        del s10[f"{top}"]

In [53]:
s10

gameId league split  playoffs                date  game  patch  \
0     1241318  KeSPA   NaN         0 2020-01-03 07:33:00     1   9.24   
1     1241322  KeSPA   NaN         0 2020-01-03 09:00:00     2   9.24   
2     1241324  KeSPA   NaN         0 2020-01-03 10:05:00     3   9.24   
3     1241328  KeSPA   NaN         0 2020-01-03 10:57:00     4   9.24   
4     1241386  KeSPA   NaN         0 2020-01-04 05:33:00     1   9.24   
...       ...    ...   ...       ...                 ...   ...    ...   
5144  1310259     EM   NaN         0 2020-08-18 20:25:00     1  10.16   
5145  1310260     EM   NaN         0 2020-08-18 20:27:00     1  10.16   
5146  1310262     EM   NaN         0 2020-08-18 20:53:00     1  10.16   
5147  1320317     EM   NaN         0 2020-08-18 20:56:00     1  10.16   
5148  1310264     EM   NaN         0 2020-08-18 21:27:00     1  10.16   

      gamelength        blue_playerid blue_ban1     blue_ban2 blue_ban3  \
0           2220       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
1           2227                   T1    Syndra       LeBlanc    Rumble   
2           1711                   T1   LeBlanc        Rumble   Lee Sin   
3           1999                   T1    Rumble         Varus   LeBlanc   
4           1812                  DRX    Qiyana       Orianna    Irelia   
...          ...                  ...       ...           ...       ...   
5144        2075               Racoon   Caitlyn         Karma       Lux   
5145        2095      SINNERS Esports      Shen         Senna      Azir   
5146        2879     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
5147        1874            G2 Arctic  Cho'Gath          Sett  Renekton   
5148        1808  Intrepid Fox Gaming     Akali       Camille     Senna   

         blue_ban4  blue_ban5  result  firstblood  firstdragon  blue_dragons  \
0            Yasuo   Renekton       1         0.0          0.0             2   
1            Braum      Leona       0         0.0          0.0             1   
2           Gragas       Olaf       1         0.0          1.0             4   
3           Aatrox     Syndra       0         0.0          1.0             2   
4       Tahm Kench  Gangplank       0         0.0          0.0             1   
...            ...        ...     ...         ...          ...           ...   
5144          Shen   Malphite       1         0.0          1.0             4   
5145  Twisted Fate    Taliyah       1         0.0          0.0             2   
5146          Olaf       Azir       1         1.0          0.0             4   
5147   Mordekaiser       Shen       1         0.0          0.0             2   
5148           Zoe    LeBlanc       1         0.0          1.0             4   

      blue_elementaldrakes  blue_infernals  blue_mountains  blue_clouds  \
0                      2.0             0.0             2.0          0.0   
1                      1.0             0.0             0.0          0.0   
2                      4.0             0.0             2.0          1.0   
3                      2.0             0.0             1.0          0.0   
4                      1.0             0.0             0.0          0.0   
...                    ...             ...             ...          ...   
5144                   4.0             2.0             1.0          0.0   
5145                   2.0             0.0             0.0          0.0   
5146                   4.0             0.0             3.0          0.0   
5147                   2.0             1.0             1.0          0.0   
5148                   4.0             0.0             2.0          1.0   

      blue_oceans  blue_elders  firstherald  blue_heralds  firstbaron  \
0             0.0          0.0          1.0           1.0         1.0   
1             1.0          0.0          0.0           0.0         0.0   
2             1.0          0.0          1.0           2.0         1.0   
3             1.0          0.0          1.0           1.0         0.0   
4 

In [54]:
bans = s10[s10.columns[s10.columns.str.contains("ban")]]

for ban in s10:
    if ban in bans:
        del s10[f"{ban}"]

In [55]:
s10

gameId league split  playoffs                date  game  patch  \
0     1241318  KeSPA   NaN         0 2020-01-03 07:33:00     1   9.24   
1     1241322  KeSPA   NaN         0 2020-01-03 09:00:00     2   9.24   
2     1241324  KeSPA   NaN         0 2020-01-03 10:05:00     3   9.24   
3     1241328  KeSPA   NaN         0 2020-01-03 10:57:00     4   9.24   
4     1241386  KeSPA   NaN         0 2020-01-04 05:33:00     1   9.24   
...       ...    ...   ...       ...                 ...   ...    ...   
5144  1310259     EM   NaN         0 2020-08-18 20:25:00     1  10.16   
5145  1310260     EM   NaN         0 2020-08-18 20:27:00     1  10.16   
5146  1310262     EM   NaN         0 2020-08-18 20:53:00     1  10.16   
5147  1320317     EM   NaN         0 2020-08-18 20:56:00     1  10.16   
5148  1310264     EM   NaN         0 2020-08-18 21:27:00     1  10.16   

      gamelength        blue_playerid  result  firstblood  firstdragon  \
0           2220       SANDBOX Gaming       1         0.0          0.0   
1           2227                   T1       0         0.0          0.0   
2           1711                   T1       1         0.0          1.0   
3           1999                   T1       0         0.0          1.0   
4           1812                  DRX       0         0.0          0.0   
...          ...                  ...     ...         ...          ...   
5144        2075               Racoon       1         0.0          1.0   
5145        2095      SINNERS Esports       1         0.0          0.0   
5146        2879     Team GamerLegion       1         1.0          0.0   
5147        1874            G2 Arctic       1         0.0          0.0   
5148        1808  Intrepid Fox Gaming       1         0.0          1.0   

      blue_dragons  blue_elementaldrakes  blue_infernals  blue_mountains  \
0                2                   2.0             0.0             2.0   
1                1                   1.0             0.0             0.0   
2                4                   4.0             0.0             2.0   
3                2                   2.0             0.0             1.0   
4                1                   1.0             0.0             0.0   
...            ...                   ...             ...             ...   
5144             4                   4.0             2.0             1.0   
5145             2                   2.0             0.0             0.0   
5146             4                   4.0             0.0             3.0   
5147             2                   2.0             1.0             1.0   
5148             4                   4.0             0.0             2.0   

      blue_clouds  blue_oceans  blue_elders  firstherald  blue_heralds  \
0             0.0          0.0          0.0          1.0           1.0   
1             0.0          1.0          0.0          0.0           0.0   
2             1.0          1.0          0.0          1.0           2.0   
3             0.0          1.0          0.0          1.0           1.0   
4             0.0          1.0          0.0          1.0           1.0   
...           ...          ...          ...          ...           ...   
5144          0.0          1.0          0.0          1.0           2.0   
5145          0.0          2.0          0.0          1.0           2.0   
5146          0.0          1.0          0.0          1.0           2.0   
5147          0.0          0.0          0.0          1.0           2.0   
5148          1.0          1.0          0.0          1.0           1.0   

      firstbaron  blue_barons  firsttower  blue_towers  firstmidtower  \
0            1.0            1         0.0            8            0.0   
1            0.0            0         0.0            2            1.0   
2            1.0            1         1.0           10            1.0   
3            0.0            0         0.0            3            0.0   
4            0.0            0         1.0            4            0.0   
.

Guardamos el dataframe en un csv, para poder cargarlo más adelante.

In [58]:
s10.to_csv("C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\Data\\process\\s10_teamstats.csv", index=False)

## Cardinalidad

In [87]:
s10

gameId league split  playoffs                date  game  patch  \
0     1241318  KeSPA   NaN         0 2020-01-03 07:33:00     1   9.24   
1     1241322  KeSPA   NaN         0 2020-01-03 09:00:00     2   9.24   
2     1241324  KeSPA   NaN         0 2020-01-03 10:05:00     3   9.24   
3     1241328  KeSPA   NaN         0 2020-01-03 10:57:00     4   9.24   
4     1241386  KeSPA   NaN         0 2020-01-04 05:33:00     1   9.24   
...       ...    ...   ...       ...                 ...   ...    ...   
5144  1310259     EM   NaN         0 2020-08-18 20:25:00     1  10.16   
5145  1310260     EM   NaN         0 2020-08-18 20:27:00     1  10.16   
5146  1310262     EM   NaN         0 2020-08-18 20:53:00     1  10.16   
5147  1320317     EM   NaN         0 2020-08-18 20:56:00     1  10.16   
5148  1310264     EM   NaN         0 2020-08-18 21:27:00     1  10.16   

      gamelength        blue_playerid  result  firstblood  firstdragon  \
0           2220       SANDBOX Gaming       1         0.0          0.0   
1           2227                   T1       0         0.0          0.0   
2           1711                   T1       1         0.0          1.0   
3           1999                   T1       0         0.0          1.0   
4           1812                  DRX       0         0.0          0.0   
...          ...                  ...     ...         ...          ...   
5144        2075               Racoon       1         0.0          1.0   
5145        2095      SINNERS Esports       1         0.0          0.0   
5146        2879     Team GamerLegion       1         1.0          0.0   
5147        1874            G2 Arctic       1         0.0          0.0   
5148        1808  Intrepid Fox Gaming       1         0.0          1.0   

      blue_dragons  blue_elementaldrakes  blue_infernals  blue_mountains  \
0                2                   2.0             0.0             2.0   
1                1                   1.0             0.0             0.0   
2                4                   4.0             0.0             2.0   
3                2                   2.0             0.0             1.0   
4                1                   1.0             0.0             0.0   
...            ...                   ...             ...             ...   
5144             4                   4.0             2.0             1.0   
5145             2                   2.0             0.0             0.0   
5146             4                   4.0             0.0             3.0   
5147             2                   2.0             1.0             1.0   
5148             4                   4.0             0.0             2.0   

      blue_clouds  blue_oceans  blue_elders  firstherald  blue_heralds  \
0             0.0          0.0          0.0          1.0           1.0   
1             0.0          1.0          0.0          0.0           0.0   
2             1.0          1.0          0.0          1.0           2.0   
3             0.0          1.0          0.0          1.0           1.0   
4             0.0          1.0          0.0          1.0           1.0   
...           ...          ...          ...          ...           ...   
5144          0.0          1.0          0.0          1.0           2.0   
5145          0.0          2.0          0.0          1.0           2.0   
5146          0.0          1.0          0.0          1.0           2.0   
5147          0.0          0.0          0.0          1.0           2.0   
5148          1.0          1.0          0.0          1.0           1.0   

      firstbaron  blue_barons  firsttower  blue_towers  firstmidtower  \
0            1.0            1         0.0            8            0.0   
1            0.0            0         0.0            2            1.0   
2            1.0            1         1.0           10            1.0   
3            0.0            0         0.0            3            0.0   
4            0.0            0         1.0            4            0.0   
.

In [ ]:
cardi = s10.nunique()
cardi

gameId              5148
league                23
split                  7
playoffs               2
date                5027
                    ... 
red_toppat10        1503
red_top_csat10        81
red_top_goldat15    2100
red_toppat15        2142
red_top_csat15       110
Length: 398, dtype: int64

In [ ]:
for i in s10:
    print(f"La cardinalidad de la variable {i} es {(s10[i].nunique()/len(s10))*100:.2f}%")

La cardinalidad de la variable gameId es 99.98%
La cardinalidad de la variable league es 0.45%
La cardinalidad de la variable split es 0.14%
La cardinalidad de la variable playoffs es 0.04%
La cardinalidad de la variable date es 97.63%
La cardinalidad de la variable game es 0.10%
La cardinalidad de la variable patch es 0.31%
La cardinalidad de la variable gamelength es 27.25%
La cardinalidad de la variable blue_playerid es 4.74%
La cardinalidad de la variable blue_ban1 es 2.08%
La cardinalidad de la variable blue_ban2 es 2.12%
La cardinalidad de la variable blue_ban3 es 2.16%
La cardinalidad de la variable blue_ban4 es 2.49%
La cardinalidad de la variable blue_ban5 es 2.41%
La cardinalidad de la variable blue_result es 0.04%
La cardinalidad de la variable blue_firstblood es 0.04%
La cardinalidad de la variable blue_firstdragon es 0.04%
La cardinalidad de la variable blue_dragons es 0.17%
La cardinalidad de la variable blue_elementaldrakes es 0.10%
La cardinalidad de la variable blue_in

In [ ]:
cols_card = []
card = []

for i in s10:
    cols_card.append(i)
    card.append((s10[i].nunique()/len(s10)*100))

p100_card = pd.DataFrame(index=cols_card, data=card)
p100_card

0
gameId            99.980579
league             0.446689
split              0.135949
playoffs           0.038842
date              97.630608
...                     ...
red_toppat10      29.190134
red_top_csat10     1.573121
red_top_goldat15  40.784618
red_toppat15      41.600311
red_top_csat15     2.136337

[398 rows x 1 columns]

In [ ]:
card_0 = pd.concat([cardi,p100_card], axis=1, )
card_0

0          0
gameId            5148  99.980579
league              23   0.446689
split                7   0.135949
playoffs             2   0.038842
date              5027  97.630608
...                ...        ...
red_toppat10      1503  29.190134
red_top_csat10      81   1.573121
red_top_goldat15  2100  40.784618
red_toppat15      2142  41.600311
red_top_csat15     110   2.136337

[398 rows x 2 columns]

In [ ]:
tipo_datos = s10.dtypes
tipo_datos

gameId               object
league               object
split                object
playoffs              int64
date                 object
                     ...   
red_toppat10        float64
red_top_csat10      float64
red_top_goldat15    float64
red_toppat15        float64
red_top_csat15      float64
Length: 398, dtype: object

In [ ]:
cardinalidad = pd.concat([card_0, tipo_datos], axis=1)
cardinalidad.columns = ["uniques", "cardinalidad", "data_type"] #añadir tipo de variable

In [ ]:
cardinalidad

uniques  cardinalidad data_type
gameId               5148     99.980579    object
league                 23      0.446689    object
split                   7      0.135949    object
playoffs                2      0.038842     int64
date                 5027     97.630608    object
...                   ...           ...       ...
red_toppat10         1503     29.190134   float64
red_top_csat10         81      1.573121   float64
red_top_goldat15     2100     40.784618   float64
red_toppat15         2142     41.600311   float64
red_top_csat15        110      2.136337   float64

[398 rows x 3 columns]

In [ ]:
cardinalidad.T

gameId    league     split  playoffs       date      game  \
uniques            5148        23         7         2       5027         5   
cardinalidad  99.980579  0.446689  0.135949  0.038842  97.630608  0.097106   
data_type        object    object    object     int64     object     int64   

                patch gamelength t1_playerid   t1_ban1   t1_ban2   t1_ban3  \
uniques            16       1403         244       107       109       111   
cardinalidad  0.31074  27.248009    4.738784  2.078073  2.116916  2.155758   
data_type     float64      int64      object    object    object    object   

              t1_ban4   t1_ban5 t1_result t1_firstblood t1_firstdragon  \
uniques           128       124         2             2              2   
cardinalidad  2.48592  2.408235  0.038842      0.038842       0.038842   
data_type      object    object     int64       float64        float64   

             t1_dragons t1_elementaldrakes t1_infernals t1_mountains  \
uniques               9                  5            5            5   
cardinalidad   0.174791           0.097106     0.097106     0.097106   
data_type         int64            float64      float64      float64   

             t1_clouds t1_oceans t1_elders t1_firstherald t1_heralds  \
uniques              5         5         3              2          3   
cardinalidad  0.097106  0.097106  0.058264       0.038842   0.058264   
data_type      float64   float64   float64        float64    float64   

             t1_firstbaron t1_barons t1_firsttower t1_towers t1_firstmidtower  \
uniques                  2         5             2        12                2   
cardinalidad      0.038842  0.097106      0.038842  0.233055         0.038842   
data_type          float64     int64       float64     int64          float64   

             t1_firsttothreetowers t1_inhibitors t2_playerid   t2_ban1  \
uniques                          2             8         244       101   
cardinalidad              0.038842       0.15537    4.738784  1.961546   
data_type                  float64       float64      object    object   

               t2_ban2   t2_ban3   t2_ban4   t2_ban5 t2_result t2_firstblood  \
uniques            105       106       135       125         2             2   
cardinalidad  2.039231  2.058652  2.621868  2.427656  0.038842      0.038842   
data_type       object    object    object    object     int64       float64   

             t2_firstdragon t2_dragons t2_elementaldrakes t2_infernals  \
uniques                   2         10                  5            5   
cardinalidad       0.038842   0.194212           0.097106     0.097106   
data_type           float64      int64            float64      float64   

             t2_mountains t2_clouds t2_oceans t2_elders t2_firstherald  \
uniques                 5         5         5         4              2   
cardinalidad     0.097106  0.097106  0.097106  0.077685       0.038842   
data_type         float64   float64   float64   float64        float64   

             t2_heralds t2_firstbaron t2_barons t2_firsttower t2_towers  \
uniques               3             2         5             2        12   
cardinalidad   0.058264      0.038842  0.097106      0.038842  0.233055   
data_type       float64       float64     int64       float64     int64   

             t2_firstmidtower t2_firsttothreetowers t2_inhibitors  \
uniques                     2                     2             8   
cardinalidad         0.038842              0.038842       0.15537   
data_type             float64               float64       float64   

             t1p1_position t1p1_player t1p1_team t1p1_champion t1p1_kills  \
uniques                  1         341       244            51         10   
cardinalidad      0.019421    6.622645  4.738784      0.990484   0.194212   
data_type           object      object    object        object      int64   

             t1p1_deaths t1p1_assists t1p1_firstblood t1p1_firstbloodkill  \
uniques        

## Limpieza